### ISSUE TITLE
Retrieve a stack container image digest from its registry and copy to status

### ISSUE THREAD

**Issue by kaczyns** [kabanero-io/kabanero-operator#437]
A stack index will contain an `image:` in its yaml which describes where that stack's container lives.  The Stack controller should attempt to look up the Sha256 (digest) of that container when the stack is activated, and copy the digest to the `status`.  

Later, during a pipeline run, a validate step in the pipeline will retrieve the digest of the container that is being used in the pipeline run, and compare it to the digests for the active stacks, to see if the pipeline run should be allowed to proceed.

This use of the digest prevents cases where the app developer has coded something like `stack:0.2` into their application, and the cluster has `stack:0.2.8` installed.  If the two share the same digest, we can allow the pipeline run to proceed.  However, if `stack:0.2.9` is available, but not installed, and it's also linked to `stack:0.2`, then the pipeline run will fail unless either the developer updates their app to use `stack:0.2.8`, or `stack:0.2.9` is installed on the cluster.

**Comment by kaczyns** [comment]
We had some more discussion about this.  The current plan is to have a status field for each stack version, called `activationDigest` which will contain the container digest at the time the stack was activated by the stack controller.  We acknowledge that the digest can change after this time, for example if someone pushes a new container using the same tag.  The thinking here is that if this happens, the validation task can recognize it (the digest for the tag at validation time was different than the tag at activation time) and can warn the user somehow.

**Comment by kaczyns** [comment]
I think to get the container image (that we're going to get the digest for) we have to concatenate the `image` field in the stack spec (version), and the actual version.  So basically un-do the stripping of the tag that we did when we created the Stack CR instance (ironic isn't it?).

We need a library to use, to go pull the SHA256 from the container registry.  One of these two is probably the right choice - the first seems more current:
https://github.com/containers/skopeo
https://github.com/google/go-containerregistry

For now we can just worry about unprotected registries.  We'll need a separate task to deal with authentication.  This also means that if for some reason we can't retrieve the digest, we need to invent a way to tell this to the user in a non-failing way.  Try to retrieve the Sha256 - if we did, great, add `activationDigest`.  If not, then we need a way to indicate that failure, and we shouldn't be retrying forever (or indicating a failure to activate the stack).
